libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncUpTarget.java

```diff
@@ -0,0 +1,200 @@
+        RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+
+        return response.isSuccess();
+    }
+
+    /**
+     * Save locally updated record back to server
+     * @param syncManager
+     * @param objectType
+     * @param objectId
+     * @param fields
+     * @return true if successful
+     * @throws JSONException
+     * @throws IOException
+     */
+    public boolean updateOnServer(SyncManager syncManager, String objectType, String objectId, Map<String, Object> fields) throws JSONException, IOException {
+        RestRequest request = RestRequest.getRequestForUpdate(syncManager.apiVersion, objectType, objectId, fields);
+        RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+
+        return response.isSuccess();
+    }
+
+    /**
+     * Fetch last modified date for a given record
+     * @param syncManager
+     * @param objectType
+     * @param objectId
+     * @return
+     * @throws JSONException
+     * @throws IOException
+     */
+    public long fetchLastModifiedDate(SyncManager syncManager, String objectType, String objectId) throws JSONException, IOException {
+        boolean isNewer = false;
+        long serverLastModified = -1L;
+        final String query = SOQLBuilder.getInstanceWithFields(Constants.LAST_MODIFIED_DATE)
+                .from(objectType)
+                .where(Constants.ID + " = '" + objectId + "'")
+                .build();
+
+        RestResponse lastModResponse = syncManager.sendSyncWithSmartSyncUserAgent(RestRequest.getRequestForQuery(syncManager.apiVersion, query));
+        if (lastModResponse != null && lastModResponse.isSuccess()) {
+            final JSONObject responseJSON = lastModResponse.asJSONObject();
+            if (responseJSON != null) {
+                final JSONArray records = responseJSON.optJSONArray(Constants.RECORDS);
+                if (records != null && records.length() > 0) {
+                    final JSONObject obj = records.optJSONObject(0);
+                    if (obj != null) {
+                        final String lastModStr = obj.optString(Constants.LAST_MODIFIED_DATE);
+                        if (!TextUtils.isEmpty(lastModStr)) {
+                            try {
+                                serverLastModified = Constants.TIMESTAMP_FORMAT.parse(lastModStr).getTime();
+                            } catch (ParseException e) {
+                                Log.e("SyncUpTarget:fetchLastModifiedDate", "Error during date parsing", e);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return serverLastModified;
+    }
+
+    /**
+     * Return ids of records to sync up
+     * @param syncManager
+     * @param soupName
+     * @return
+     */
+    public Set<String> getIdsOfRecordsToSyncUp(SyncManager syncManager, String soupName) throws JSONException {
```

Comment:
Yeah, seems better that the sync up target owns the records, like the sync down target does. :+1: